src/XrdTls/XrdTlsTempCA.hh

```diff
@@ -0,0 +1,113 @@
+/******************************************************************************/
+/*                                                                            */
+/*               X r d T l s T e m p C A . h h                                */
+/*                                                                            */
+/* (c) 2021 by the Board of Trustees of the Leland Stanford, Jr., University  */
+/*   Produced by Brian Bockelman                                              */
+/*                                                                            */
+/* This file is part of the XRootD software suite.                            */
+/*                                                                            */
+/* XRootD is free software: you can redistribute it and/or modify it under    */
+/* the terms of the GNU Lesser General Public License as published by the     */
+/* Free Software Foundation, either version 3 of the License, or (at your     */
+/* option) any later version.                                                 */
+/*                                                                            */
+/* XRootD is distributed in the hope that it will be useful, but WITHOUT      */
+/* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or      */
+/* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public       */
+/* License for more details.                                                  */
+/*                                                                            */
+/* You should have received a copy of the GNU Lesser General Public License   */
+/* along with XRootD in a file called COPYING.LESSER (LGPL license) and file  */
+/* COPYING (GPL license).  If not, see <http://www.gnu.org/licenses/>.        */
+/*                                                                            */
+/* The copyright holder's institutional names and contributor's names may not */
+/* be used to endorse or promote products derived from this software without  */
+/* specific prior written permission of the institution or contributor.       */
+/******************************************************************************/
+
+#include <atomic>
+#include <string>
+#include <memory>
+
+// Forward dec'ls.
+class XrdSysError;
+
+namespace XrdTls {
```

Comment:
We don't use namespaces in this way. If the class is named "XrdTls<whatever>" then it already has placed itself in an implicit flat namespace. If a class that would have been in a namespace, say XrdTls, it won't call itself XrdTls<whatever> but simply <whatever>. This avoids  redundant references to XrdTls::XrdTls<whatever> which is rather annoying. No, you cannot use "using" to avoid that (though obviously it does avoid it). I suggest simply not putting the class in the *new* XrdTls namespace as it's simply not needed and actually makes it inconvenient to debug.

Reply:
Ok, will collapse this into the main namespace.  Easy enough.